### DISCUSSION TITLE
Compiling Less styles

### DISCUSSION THREAD

**Discussion by picarda** [searxng/searxng#1645] [Q&A] [answered]
Hello, how to compile the less styles?
I did install NPM... then run $make styles, but got an error...
What I'm missing here?

Thank you.

**Comment by return42** [comment, marked as answer]
For development I recommend to install packages needed by a buildhost, [documented here](https://docs.searxng.org/admin/buildhosts.html)

    sudo -H ./utils/searxng.sh install buildhost

> I did install NPM... then run $make styles, but got an error...

You don't need to install NPM on your OS. Beside from the *buildhost* packages from above, the SearXNG developer environment is build up in the `./local` and `./nvm` folder (independent from the OS packages).

To get a node environment (in `./nvm`) use:

    make node.env

To build & commit the /static files (including less styles) use:

    make static.build.commit

But first read [Development Quickstart](https://docs.searxng.org/dev/quickstart.html) carefully :+1:

**Reply by picarda** [reply]
Thank you!
